src/org/opendatakit/briefcase/transfer/TransferForms.java

```diff
@@ -206,6 +206,10 @@ public String getLastCursor(FormStatus fs) {
     return Optional.ofNullable(lastPullCursorsByFormId.get(fs.getFormId())).orElse("");
   }
 
+  public void setLastPullCursor(FormStatus fs, String cursor) {
+    lastPullCursorsByFormId.put(fs.getFormId(), cursor);
```

Comment:
Why not have the form ID as the formal parameter rather than have this method concerned with a FormStatus?

Reply:
I guess you're right, but I'm struggling to not think about this until we can start working on a central database. All the domain is inside out. The last pull cursor should be the "Form"'s concern...